### ISSUE TITLE
Trigger loop introduced by the new fuel-based reveal

### ISSUE THREAD

**Issue by parno** [Microsoft/dafny#9]
In the Dafny code below, the latest version of Dafny times out.  It will succeed if you put the fuel for `Looper` one notch higher (as in the commented code below).  Eliminating the call to reveal_Looper() also allows verification to succeed.  

```
function Transform(x:int) : int

lemma TransformProperties()
    ensures forall x1, x2 {:trigger Transform(x1), Transform(x2)} :: Transform(x1) == Transform(x2) ==> x1 == x2;       

function {:opaque} Looper(input:seq<int>) : seq<int>
    ensures |Looper(input)| == |input|;
    ensures forall i :: 0 <= i < |input| ==> Looper(input)[i] == Transform(input[i])
{
    if |input| == 0 then []
    else [Transform(input[0])] + Looper(input[1..])
}

//lemma {:fuel Looper,2} proof(s1:seq<int>, s2:seq<int>)  // Succeeds
lemma proof(s1:seq<int>, s2:seq<int>)
    requires Looper(s1) == Looper(s2);
    ensures forall i :: i in s1 <==> i in s2;
{
    reveal_Looper();
    TransformProperties();
}
```

Inside the proof lemma, the new Dafny sets the fuel for all opaque functions to their respective base values via statements like:

```
assume StartFuel_F = Base_F
assume StartFuelAssert_F = Base_F
```
The reveal call then says:
```
assume StartFuel_F = LS(MoreFuel_F1)
assume StartFuelAssert_F = LS(LS(MoreFuel_F1))
```
for a new constant `MoreFuel_F1`.

This then gives us:
```
                LS(MoreFuel_F1) == LS(LS(MoreFuel_F1))
                ==> MoreFuel_F1 == LS(MoreFuel_F1)
```
This produces a trigger loop of infinite fuel, which produced an impressive 2.5 million instantiations in 30 seconds.

It succeeds when we give it more fuel because we then get:
```
StartFuel_F = LS(LS(Base_F))
StartFuelAssert_F = LS(LS(LS(Base_F)))

StartFuel_F = LS(MoreFuel_F1)
StartFuelAssert_F = LS(LS(MoreFuel_F1))
```
There are a couple of potential solutions here.  My current proposal is that we have reveal instead emit:
```
StartFuel_F = LS(MoreFuel_F1)
StartFuelAssert_F = LS(LS(MoreFuelAssert_F1))
```
In other words, use a different constant for the `StartFuel` and the `StartFuelAssert`.  That should prevent this kind of loop (and indeed works for this example).
